File: sdk/typescript/src/rs485.ts
import type { WlteClient } from './client.js'
import type { ApiEnvelope, CommandExecution, RS485BaudRateOptions, RS485TransceiveOptions } from './types.js'

export class RS485Api {
  constructor(private readonly client: WlteClient) {}

  async transceive(deviceId: string, options: RS485TransceiveOptions): Promise<CommandExecution> {
    const response = await this.client.request<ApiEnvelope<CommandExecution>>(
      `/wlte/v1/devices/${encodeURIComponent(deviceId)}/rs485/transceive`,
      {
        method: 'POST',
        headers: {
          'Idempotency-Key': options.idempotencyKey,
        },
        body: {
          requestHex: options.requestHex,
        },
      },
    )
    return response.data
  }

  async setBaudRate(deviceId: string, options: RS485BaudRateOptions): Promise<CommandExecution> {
    const response = await this.client.request<ApiEnvelope<CommandExecution>>(
      `/wlte/v1/devices/${encodeURIComponent(deviceId)}/rs485/baud-rate`,
      {
        method: 'PUT',
        headers: {
          'Idempotency-Key': options.idempotencyKey,
        },
        body: {
          baudRate: options.baudRate,
        },
      },
    )
    return response.data
  }
}

File: sdk/go/rs485.go
package wlteopenapi

import "context"

type RS485Service struct {
	client *Client
}

func (s *RS485Service) Transceive(ctx context.Context, deviceID string, options RS485TransceiveOptions) (Command, error) {
	var result Command
	err := s.client.request(
		ctx,
		"POST",
		"/wlte/v1/devices/"+urlEscape(deviceID)+"/rs485/transceive",
		nil,
		map[string]string{"Idempotency-Key": options.IdempotencyKey},
		map[string]string{"requestHex": options.RequestHex},
		&result,
	)
	return result, err
}

func (s *RS485Service) SetBaudRate(ctx context.Context, deviceID string, options RS485BaudRateOptions) (Command, error) {
	var result Command
	err := s.client.request(
		ctx,
		"PUT",
		"/wlte/v1/devices/"+urlEscape(deviceID)+"/rs485/baud-rate",
		nil,
		map[string]string{"Idempotency-Key": options.IdempotencyKey},
		map[string]int{"baudRate": options.BaudRate},
		&result,
	)
	return result, err
}
